src/azure_devtools/scenario_tests/base.py

```diff
@@ -142,6 +143,11 @@ def tearDown(self):
         os.environ = self.original_env
 
     def _process_request_recording(self, request):
+        # URL parsing fix to account for '//' vs '/' in different versions of python
+        # - Python 2.7 for 2.7/3.3/3.4 (...Microsoft.Compute//availabilitySets...)
+        # - Python 3.5 (...Microsoft.Compute/availabilitySets...)
+        request.uri = re.sub('(?<!:)//', '/', request.uri)
+
```

Comment:
Please put this in a separate recording processor and add it to the other recording processors. The actual logic of altering the recording should not be placed in the framework.